### ISSUE TITLE
Equation rendering

### ISSUE THREAD

**Issue by PascalGitz** [gutow/Algebra_with_Sympy#26]
> Thank you for incorporating it. The approach, as shown in the following snippet, cannot be rendered in LaTeX. But it works fine in the Jupyternotebook.
> 
> ```
> $$
> \\begin{equation}\\frac{a}{b}=c\\end{equation}
> $$
> ```

@PascalGitz I set it up to generate Latex that renders properly in Jupyter notebooks. Because of the way strings are processed in the notebook backslashes need to be escaped with an extra backslash to appear in the notebook. I may be able to do some fancy footwork with Python raw strings to fix this. However, I need to know what situations should provide output with single rather than doubled backslashes. Can you provide some direction on that? If you can, please open a new issue with the information.

_Originally posted by @gutow in https://github.com/gutow/Algebra_with_Sympy/issues/22#issuecomment-1875974698_

**Comment by PascalGitz** [comment]
I agree with the double backslashes. But having `\\begin{equation} \\end{equation}` in a double dollarsign environment does the trick two times.

Maybe something like this snippet would help? It renders it properly in the notebook, aswell as in the LaTeX output.
```
from IPython.display import Markdown
display(Markdown('\\begin{equation}\\frac{a}{b}=c\\end{equation}'))
```

I'm sorry for not testing your code beforehand.

**Comment by gutow** [comment]
@PascalGitz I do not think I understand how you are using this to create the LaTex output. What process are you calling that balks at the extra backslashes? Please provide a simple step-by-step sequence I can follow to reproduce the problem. Then I can try to figure out how to make it work across command-line, IPython, Jupyter notebooks, and what you are doing.

**Comment by gutow** [comment]
@PascalGitz I did try the Jupyter lab menu item `File>Save and Export Notebook as...>LaTex` and could not reproduce the issue of extra backslashes.

**Comment by PascalGitz** [comment]
I see the problem lies on Quartos side. It converts the double dollar signs to '\\[' which then results in the problem. You can reproduce the error by using this snippet. Assuming you have quarto installed.

```
from IPython.display import Markdown
display(Markdown('$$\\begin{equation}\int_0^1 x^2 dx \\end{equation}$$'))

!quarto render test.ipynb --to pdf
```

**Comment by gutow** [comment]
I think I can fix this behavior, but have to make and test a number of rewrites to the output string construction. It looks like I overlooked the fact that Markdown strings of the form `\begin{equation}...\end{equation}` will always be recognized as LaTex, independently of whether it is surrounded by the `$...$` markers. Other ways of generating .pdf and .tex from notebooks are happy with either `\begin{equation}...\end{equation}` or `$\begin{equation}...\end{equation}$`. Since they appear to accept both, I believe it is safe to use only the first form, which Quarto appears to accept.

**Comment by gutow** [comment]
@PascalGitz 
Please try this development release, which I believe fixes the problem.
`pip install Algebra-with-SymPy==1.0.1.dev1`

**Comment by PascalGitz** [comment]
@gutow 

```
import algebra_with_sympy as spa # Automatically imports sympy into the namespace `spa`
from sympy import Integer
from sympy.abc import *

spa.algwsym_config.output.latex_as_equations = True

spa.Eqn(x, 20*y)
```

I am using your dev1 version. It does not get rendered as an Equation. Am I configuring something wrong?

**Comment by gutow** [comment]
> @gutow
> 
> ```
> import algebra_with_sympy as spa # Automatically imports sympy into the namespace `spa`
> from sympy import Integer
> from sympy.abc import *
> 
> spa.algwsym_config.output.latex_as_equations = True
> 
> spa.Eqn(x, 20*y)
> ```
> 
> I am using your dev1 version. It does not get rendered as an Equation. Am I configuring something wrong?

Try `from algebra_with_sympy import *`. That should configure things properly. There is no need to import sympy separately. That is done automatically by algebra_with_sympy. You still need to import from `sympy.abc` to have all the symbols defined for you.

**Comment by PascalGitz** [comment]
it works, great job. thank you

**Comment by gutow** [comment]
> it works, great job. thank you

Glad to hear it. Keep using that development version until I release 1.0.1. There are some additional fixes (such as documentation for how to use the settings when importing as a module) that need to go in before I release it.

**Comment by gutow** [comment]
fixed in v1.0.1
